test/test.go

```diff
@@ -0,0 +1,213 @@
+package test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/bitrise-io/bitrise/models"
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/steps-deploy-to-bitrise-io/test/converters"
+	"github.com/google/uuid"
+)
+
+func findImages(testDir string) (imageFilePaths []string) {
+	for _, ext := range []string{".jpg", ".jpeg", ".png"} {
+		if paths, err := filepath.Glob(filepath.Join(testDir, "*"+ext)); err == nil {
+			imageFilePaths = append(imageFilePaths, paths...)
+		}
+		if paths, err := filepath.Glob(filepath.Join(testDir, "*"+strings.ToUpper(ext))); err == nil {
+			imageFilePaths = append(imageFilePaths, paths...)
+		}
+	}
+	return
+}
+
+// ResultAssets ...
+type ResultAssets struct {
+	XMLContent []byte                    `json:"xml"`
+	ImagePaths []string                  `json:"-"`
+	StepInfo   models.TestResultStepInfo `json:"step_info"`
+}
+
+// Bundle ...
+type Bundle []ResultAssets
+
+// ParseTestResults ...
+func ParseTestResults(testsRootDir string) (bundle Bundle, err error) {
+	// read dirs in base tests dir
+	testDirs, err := ioutil.ReadDir(testsRootDir)
+	if err != nil {
+		return nil, err
+	}
+
+	// find test results in each dir, skip if invalid test dir
+	for _, testDir := range testDirs {
+		testDirPath := filepath.Join(testsRootDir, testDir.Name())
+
+		// read one level of file set only <root_tests_dir>/<test_dir>/file_to_get
+		testFiles, err := filepath.Glob(filepath.Join(testDirPath, "*"))
+		if err != nil {
+			return nil, err
+		}
+
+		// find step-info in dir
+		var stepInfo *models.TestResultStepInfo
+		for _, file := range testFiles {
+			if strings.HasSuffix(file, "step-info.json") {
+				stepInfoFileContent, err := fileutil.ReadBytesFromFile(file)
+				if err != nil {
+					return nil, err
+				}
+				if err := json.Unmarshal(stepInfoFileContent, &stepInfo); err != nil {
+					return nil, err
+				}
+			}
+		}
+
+		// if no step-info.json then skip
+		if stepInfo == nil {
+			continue
+		}
+
+		// get the handler that can manage test type contained in the dir
+		for _, handler := range converters.List() {
+			handler.SetFiles(testFiles)
+			// skip if couldn't find handler for content type
+			if handler.Detect() {
+				xml, err := handler.XML()
+				if err != nil {
+					return nil, err
+				}
+
+				// so here I will have image paths, xml data, and step info per test dir in a bundle info
+				bundle = append(bundle, ResultAssets{
+					XMLContent: xml,
+					ImagePaths: findImages(filepath.Join(testsRootDir, testDir.Name())),
+					StepInfo:   *stepInfo,
+				})
+			}
+		}
+	}
+	return bundle, nil
+}
+
+func checkResponse(resp *http.Response) error {
+	if resp.StatusCode < 200 || 299 < resp.StatusCode {
```

Comment:
What about:
`if resp.StatusCode < http.StatusOK || http. StatusIMUsed < resp.StatusCode`

More about it:
https://golang.org/pkg/net/http/

---

BTW: I think it will be enough to check for 
- StatusOK                   = 200
- StatusCreated           = 201

Reply:
You are right, however not our team that writes the server to be able to use it. I was ware of the http constants however this way it is more easily visible that we are capturing status codes between an interval.